src/main/java/org/opensearch/timeseries/constant/CommonMessages.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.timeseries.constant;
+
+import java.util.Locale;
+
+public class CommonMessages {
+    // ======================================
+    // Validation message
+    // ======================================
+    public static String NEGATIVE_TIME_CONFIGURATION = "should be non-negative";
+    public static String INVALID_RESULT_INDEX_MAPPING = "Result index mapping is not correct for index: ";
+    public static String EMPTY_NAME = "name should be set";
+    public static String NULL_TIME_FIELD = "Time field should be set";
+    public static String EMPTY_INDICES = "Indices should be set";
+
+    public static String getTooManyCategoricalFieldErr(int limit) {
+        return String.format(Locale.ROOT, CommonMessages.TOO_MANY_CATEGORICAL_FIELD_ERR_MSG_FORMAT, limit);
+    }
+
+    public static final String TOO_MANY_CATEGORICAL_FIELD_ERR_MSG_FORMAT = "We can have only %d categorical field/s.";
+    public static String CAN_NOT_FIND_RESULT_INDEX = "Can't find result index ";
+    public static String INVALID_CHAR_IN_RESULT_INDEX_NAME =
+        "Result index name has invalid character. Valid characters are a-z, 0-9, -(hyphen) and _(underscore)";
+    public static String FAIL_TO_VALIDATE = "fail to validate";
+    public static String INVALID_TIMESTAMP = "Timestamp field: (%s) must be of type date";
+    public static String NON_EXISTENT_TIMESTAMP = "Timestamp field: (%s) is not found in index mapping";
+    public static String INVALID_NAME = "Valid characters for name are a-z, A-Z, 0-9, -(hyphen), _(underscore) and .(period)";
+    // change this error message to make it compatible with old version's integration(nexus) test
+    public static String FAIL_TO_FIND_CONFIG_MSG = "Can't find config with id: ";
+    public static final String CAN_NOT_CHANGE_CATEGORY_FIELD = "Can't change category field";
+    public static final String CAN_NOT_CHANGE_CUSTOM_RESULT_INDEX = "Can't change custom result index";
+    public static final String CATEGORICAL_FIELD_TYPE_ERR_MSG = "A categorical field must be of type keyword or ip.";
+    // Modifying message for FEATURE below may break the parseADValidationException method of ValidateAnomalyDetectorTransportAction
+    public static final String FEATURE_INVALID_MSG_PREFIX = "Feature has an invalid query";
+    public static final String FEATURE_WITH_EMPTY_DATA_MSG = FEATURE_INVALID_MSG_PREFIX + " returning empty aggregated data: ";
+    public static final String FEATURE_WITH_INVALID_QUERY_MSG = FEATURE_INVALID_MSG_PREFIX + " causing a runtime exception: ";
```

Comment:
"caused a runtime" ... but it can be "causing. .." if there are ongoing retries.

Reply:
Frontend code used this message in its test: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/710e3c4ddadf3689b7f014b5df88f2a82b8db999/public/pages/ReviewAndCreate/containers/__tests__/ReviewAndCreate.test.tsx#L195-L198

Also, there is a comment above "// Modifying message for FEATURE below may break the parseADValidationException method of ValidateAnomalyDetectorTransportAction". Can I keep it as it is for now?

Reply:
Yes - there is no need to break other artefacts!